modules/model/src/main/scala/io/janstenpickle/trace4cats/model/SpanId.scala

```diff
@@ -1,28 +1,52 @@
 package io.janstenpickle.trace4cats.model
 
-import cats.syntax.functor._
+import cats.effect.kernel.Sync
 import cats.syntax.show._
-import cats.{Eq, Functor, Show}
-import io.janstenpickle.trace4cats.model.random.Random
+import cats.{Eq, Show}
 import org.apache.commons.codec.binary.Hex
 
+import java.util.concurrent.ThreadLocalRandom
 import scala.util.Try
 
 case class SpanId private (value: Array[Byte]) extends AnyVal {
   override def toString: String = show"SpanId($this)"
 }
 
 object SpanId {
-  def apply[F[_]: Functor: Random]: F[SpanId] =
-    Random[F].nextBytes(8).map(new SpanId(_))
+  val size = 8
+
+  trait Gen[F[_]] {
+    def gen: F[SpanId]
+  }
+
+  object Gen {
+    def apply[F[_]](implicit ev: Gen[F]): Gen[F] = ev
+
+    def from[F[_]](_gen: F[SpanId]): Gen[F] = new Gen[F] {
+      def gen: F[SpanId] = _gen
+    }
+
+    implicit def threadLocalRandomSpanId[F[_]: Sync]: Gen[F] = Gen.from(Sync[F].delay {
```

Comment:
I assume that `cats.effect.std.Random` is not an option here (though it expresses the capability of generating some random bytes way better than `Sync` because it's not in cats-effect-kernel?

Other than that, the PR looks good to me 👍

Reply:
I intentionally didn't use `Random` as a bound on `F[_]` since it doesn't have auto derivation from `Sync`.
I could summon `Random` instance within my implementation, but that's not even shorter and brings `cats.effect.std` dependency.